frontend/javascripts/dashboard/folders/details_sidebar.tsx

```diff
@@ -79,6 +84,26 @@ function getMaybeSelectMessage(datasetCount: number) {
 function DatasetDetails({ selectedDataset }: { selectedDataset: APIDatasetCompact }) {
   const context = useDatasetCollectionContext();
   const { data: fullDataset, isFetching } = useDatasetQuery(selectedDataset);
+  const activeUser = useSelector((state: OxalisState) => state.activeUser);
+  const owningOrganizationDisplayName = useFetch(
+    async () => {
+      const owningOrga = await getOrganization(selectedDataset.owningOrganization);
```

Comment:
this request will always be triggered now when the the sidebar changes because the user clicked on a different dataset. this is done even when the organization is not needed (a) because the dataset is owned by the active organization anyway and (b) because it was already fetched before.
I think (a) is fine when we add caching anyway, since it will only mean one additional request (for the owning organization) for the whole lifetime of the tab.
Therefore, I suggest to add caching. The library react-query can be very useful for this. It would look something like this:

```js
const { data: owningOrganization } = useQuery(
    ["organizations", selectedDataset.owningOrganization],
    () => getOrganization(selectedDataset.owningOrganization),
    {
        refetchOnWindowFocus: false,
    }
);
```

the first parameter tells react-query under which key the result should be cached. let me know if you have questions.

this will only work in the details_sidebar here, I think, because the dataset info tab doesn't use functional components. however, in the dataset info tab, the fetch is not problematic, since the active dataset doesn't really change over time.

Reply:
makes sense, thank you!

Reply:
this might be a stupid question but I am trying to understand why you propose `organizations` as the key for the cache. does the cache contain all organizations that have ever been cached? or only one at a time?

Reply:
it's not a stupid question :) they key is a composite of the string `"organizations"` and the actual id `selectedDataset.owningOrganization`. this means, all organizations are cached by using their respective ids. the string "organizations" is simply there to group all cached organizations together. the react-query cache is global and can be used for other stuff, too (e.g., datasets). that's why, we establish a "cache hierarchy" with the string.